datalad/support/annexrepo.py

```diff
@@ -699,6 +699,16 @@ def is_remote_annex_ignored(self, remote):
             default=False
         )
 
+    def set_remote_ignore(self, remote, ignored=False):
+        """Add or set an annex-ignore section for a given remote"""
+        # check if there is a section already
+        if self.config.has_section('remote.{}.annex-ignore'.format(remote)):
+            # overwrite (set) config
+            self.config.set('remote.{}.annex-ignore'.format(remote), ignored, scope='local')
+        else:
+            # add it
+            self.config.add('remote.{}.annex-ignore'.format(remote), ignored, scope='local')
```

Comment:
`.set` seems to work just fine for unknown yet options, even if the entire section doesn't exist
```
(git-annex)lena:~/datalad/openneuro/ds000001[master]git
$> python3 -c 'import datalad.api as dl; print(dl.Dataset(".").config.set("remote.origin.brand-new", "true", scope="local"))'
None

$> git config remote.origin.brand-new                                                                                        
true

$> python3 -c 'import datalad.api as dl; print(dl.Dataset(".").config.set("remote.unknown.brand-new", "true", scope="local"))' 
None

$> git config remote.unknown.brand-new                                                                                  
true
```
but how would we "unset" or remove that setting entirely -- e.g. if I would prefer to leave it for git-annex to *possibly* set. I.e. may be if the `ignored` is `None` it would remove that setting if it is set (to any - true or false)?